clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinatorTest.java

```diff
@@ -260,6 +261,45 @@ public void testJoinGroupRequestWithMemberIdMisMatch() {
         assertFalse(future.isRetriable());
     }
 
+    @Test
+    public void testSyncGroupRequestWithFencedInstanceIdException() {
+        setupCoordinator();
+        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+
+        final String memberId = "memberId";
+        final int generation = -1;
+
+        mockClient.prepareResponse(joinGroupFollowerResponse(generation, memberId, JoinGroupResponse.UNKNOWN_MEMBER_ID, Errors.NONE));
+        mockClient.prepareResponse(syncGroupResponse(Errors.FENCED_INSTANCE_ID));
+        try {
+            coordinator.ensureActiveGroup();
+            fail("Failed to catch expected FencedInstanceIdException");
+        } catch (RuntimeException e) {
+            assertTrue("Sync group request was fenced due to duplicate instance id", e instanceof FencedInstanceIdException);
+        }
+    }
+
+    @Test
+    public void testHeartbeatRequestWithFencedInstanceIdException() {
+        setupCoordinator();
+        mockClient.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+
+        final String memberId = "memberId";
+        final int generation = -1;
+
+        mockClient.prepareResponse(joinGroupFollowerResponse(generation, memberId, JoinGroupResponse.UNKNOWN_MEMBER_ID, Errors.NONE));
+        mockClient.prepareResponse(syncGroupResponse(Errors.NONE));
+        mockClient.prepareResponse(heartbeatResponse(Errors.FENCED_INSTANCE_ID));
+
+        coordinator.ensureActiveGroup();
+
+        RequestFuture<Void> future = coordinator.sendHeartbeatRequest();
```

Comment:
I think `ensureActiveGroup` will start the heartbeat thread. Could we rely on it to send the heartbeat request? For example, I think we can sleep for the heartbeat interval and then call `pollHeartbeat` in a loop to check for the failure. That would exercise the catch logic in the heartbeat thread.